### DISCUSSION TITLE
Help choosing Google Coral TPU

### DISCUSSION THREAD

**Discussion by avvalex** [blakeblackshear/frigate#8279] [General]
Hi!
I'm using an Asrock J5005 motherboard (https://www.asrock.com.tw/mb/Intel/J5005-ITX/index.it.asp) and i would like to add a Coral in order to externd object recognition to all my cams.
I'm running Frigate in Docker.
I know that the USB version would be the easiest to implement, but i don't like the lack of thermal throttling and the fact that it becomes too hot.
My mobo has a PCI express 1x port and a M2 port for wifi.....

Do you think i can put a "M.2 Accelerator A+E key" or even a "M.2 Accelerator with Dual Edge TPU" (is it possible to make the second TPU work?)???

Or should i buy a "Mini PCIe Accelerator" with an adaptor (which one?)??

Many thanks!!!!

**Comment by psCLE** [comment]
I recently installed a Coral M.2 A+E accelerator (G650-04527-01) in a Dell M.2 "wifi" slot running Debian 12. My Frigate is a Docker compose container. With the Dell, I had to disable secure boot in order for Debian to recognize the TPU. I don't know enough about this to know if this is the best combination but I wanted an internal only solution, not a USB dongle.

**Reply by psCLE** [reply]
"wifi" is in quotes because the M.2 slot on the logic board is labeled WIFI.

**Comment by jubjubrsx** [comment]
just my 0.02$ but if I were you I'd still try to find a B+M key and get the pci-e adapter from amazon (like 7-10$) its an added costs yes, but if you ever want to upgrade, good chance the new mobo wont have a A+E key which if you were to get an adapter for that they are pricey, while the pci-e adapter that will be in the new pc for sure.

**Reply by avvalex** [reply]
So you think it's better a Coral B+M with an adapter to PCI Express 2.0 x1? (could you link me a working adapte please?)
And why not a MINI PCI-E Coral TPU with an adapter MINI-PCI-E to PCI-E?

**Comment by jubjubrsx** [comment]
sorry for the late reply been sick with covid
I picked up this https://www.amazon.com/gp/product/B07R9VB35J/ref=ppx_yo_dt_b_search_asin_title?ie=UTF8&th=1

and the m.2 from seeedstudio. I just say the B+m is more common, my motherboard has 2x B+M keys, and you can find adapters for 4x B+M..

the A+E key is generally just for wifi/nic's which most mobo's have it built on so they dont have the slots anymore.

**Comment by rodrigoratan** [comment]
I purchased the M2 Coral last week from China (it didnt arrive yet) and didn't know it needs an adapter. I was thinking I only needed to plug in the M2 slot. It wont work?

**Reply by jubjubrsx** [reply]
depends on your motherboard, and which key you got (a+e or b+m)

**Reply by rodrigoratan** [reply]
only B+E :-( 

I purchased now this https://pt.aliexpress.com/item/1005005225206872.html
